### ISSUE TITLE
3.0 - Discounts: Discrepancies in naming for product requirements and exclusions 

### ISSUE THREAD

**Issue by cklosowski** [easydigitaldownloads/easy-digital-downloads#7348]
While trying to fix some issues with the unit tests, I noticed we have some real confusing situations around discount code product requirements and product exclusions.

Product Requirements:
* The form fields for add/update discount code use `product_reqs` as the name.
* The meta key is `product_requirement` (notice this is singular, and fully spelled out)
* The `EDD_Discount` object uses a property of `product_reqs`, which while matching the POST values, does not match the meta key.
* The helper function to get the product requirements is named `edd_get_discount_product_reqs`.

Product Exclusions:
* The form fields for add/update discount code use `excluded_products` as the name.
* The meta key is `excluded_product` (notice this is singular)
* The `EDD_Discount` object uses a property of `excluded_products`, which matches POST data but not the meta key.
* The helper function to get the product requirements is named `edd_get_discount_excluded_products` (plural, but again ultimately getting the data from key that is singular)

I feel like we should really focus on unifying these, with full backwards compatibility from what 2.9 uses `product_reqs` and `excluded_products` for the `EDD_Discount` properties. We'd need to build in some backwards compatibility to get the data into a commonly used name that matches the meta key (which I think is the ideal situation so that we have the same name all over the place).

**Comment by cklosowski** [comment]
@mintplugins would love your feedback here.

**Comment by mintplugins** [comment]
@cklosowski Was 2.9 consistently calling them `product_reqs` and `excluded_products`?

**Comment by cklosowski** [comment]
Yes, it was, I want to stop using the shortened ones, and also update the meta key so that they all match. I would like to see and use:
`product_requirements` and `excluded_products` everywhere.

So we'll have to update the add/edit discount form to use these, the saving of them, the `edd_add_discount` and `edd_update_discount` functions (which I'm ok with, just need to do a bit of backcompat).

We'll have to update the meta keys with backwards compat.

And we'll update the `EDD_Discount` properties, with backwards compat.

**Comment by mintplugins** [comment]
Sounds like a good approach to me
